apollo-api/src/main/java/com/spotify/apollo/RequestValue.java

```diff
@@ -57,24 +58,39 @@ private static Request create(
       Optional<ByteString> payload) {
     return new AutoValue_RequestValue(
         method, uri,
-        parameters,
-        headers,
+        ImmutableMap.copyOf(parameters),
+        ImmutableMap.copyOf(headers),
         service,
         payload);
   }
 
+  @Override
+  public Request withUri(String uri) {
+    return create(method(), uri, parameters(), headers(), service(), payload());
```

Comment:
Should perhaps the parameters be cleared here?

Reply:
Rebuild from new URL